motion_computation/src/motion_computation_worker.cpp

```diff
@@ -129,4 +138,218 @@ void MotionComputationWorker::setConfidenceDropRate(double drop_rate)
   prediction_confidence_drop_rate_ = drop_rate;
 }
 
+void MotionComputationWorker::setExternalObjectPredictionMode(int external_object_prediction_mode)
+{
+  external_object_prediction_mode_ = static_cast<MotionComputationMode>(external_object_prediction_mode);
+}
+
+void MotionComputationWorker::geoReferenceCallback(const std_msgs::String& georef)
+{
+  if (georeference_ == georef.data)
+    return;
+  else
+  {
+    georeference_ = georef.data;
+  }
+  local_projector_ = std::make_shared<lanelet::projection::LocalFrameProjector>(lanelet::projection::LocalFrameProjector(georeference_.c_str()));
+}
+
+void MotionComputationWorker::mobilityPathCallback(const cav_msgs::MobilityPath& msg)
+{
+  mobility_path_list_.objects.push_back(mobilityPathToExternalObject(msg));
+  if (external_object_prediction_mode_ == MOBILITY_PATH_ONLY)
+  {
+    obj_pub_(mobility_path_list_);
+    // Clear mobility msg path queue since it is published
+    mobility_path_list_.objects = {};
+    return;
+  }
+}
+
+cav_msgs::ExternalObject MotionComputationWorker::mobilityPathToExternalObject(const cav_msgs::MobilityPath& msg)
+{
+  cav_msgs::ExternalObject output;
+
+  if (!local_projector_)
+  {
+    ROS_WARN_STREAM("Motion computation has not received georeference string yet!");
+    return output;
+  }
+
+  // get reference origin in ECEF (convert from cm to m)
+  double ecef_x = (double)msg.trajectory.location.ecef_x/100.0;
+  double ecef_y = (double)msg.trajectory.location.ecef_y/100.0;
+  double ecef_z = (double)msg.trajectory.location.ecef_z/100.0;
+
+  // Convert general information
+  for (size_t i = 0; i < msg.header.sender_bsm_id.size(); i+=2) // convert hex std::string to uint8_t array
+  {
+    int num = 0;
+    sscanf(msg.header.sender_bsm_id.substr(i, i + 2).c_str(), "%x", &num);
+    output.bsm_id.push_back((uint8_t)num);
+  }
+  // first point's timestamp
+  output.header.stamp = ros::Time((double)msg.header.timestamp/ 1000.0);
+
+  // If it is a static object, we finished processing
+  if (msg.trajectory.offsets.size() < 2)
+  {
+    output.dynamic_obj = false;
+    return output;
+  }
+  output.dynamic_obj = true;
+
+  // get planned trajectory points
+  auto prev_pt_msg = msg.trajectory.offsets[0]; // setup first point to be processed later
+  cav_msgs::PredictedState prev_state;
+  lanelet::BasicPoint3d prev_pt_ecef {ecef_x + (double)prev_pt_msg.offset_x /100.0, ecef_y + (double)prev_pt_msg.offset_y /100.0, ecef_z + (double)prev_pt_msg.offset_z /100.0};
+
+  auto prev_pt_map = local_projector_->projectECEF(prev_pt_ecef, -1);
```

Comment:
Adding lanelet2_extension and lanelet2 as a dep seems excessive just to get this functionality. 
The "earth" frame is already ECEF, so if you just lookup and cache the earth->map transform once you can apply it to all the points without depending on lanelet2.  If you think the current approach is cleaner though I'm open to an explanation of that.